src/main/java/seedu/address/model/internship/InternshipItem.java

```diff
@@ -165,6 +167,19 @@ public void setRequirements(Set<Requirement> requirements) {
         this.requirements = requirements;
     }
 
+    /**
+     * Returns true if the skill list provided matches this internship item. A match is found when any skill in
+     * the skill list contains an InternshipItem's requirement.
+     *
+     * @param skillList List of skills to check.
+     * @return True if the skill list provided matches the internship item, false otherwise.
+     */
+    public boolean matches(List<String> skillList) {
+        assert skillList != null;
+        return requirements.stream().anyMatch(requirement ->
+                skillList.stream().anyMatch(skill -> skill.contains(requirement.toString())));
```

Comment:
honestly I don't think this is a good idea?
suppose the requirement is "c" a common programming language, it will literally match with any word with a letter c (like cooking) or java with javascript, etc.

Reply:
@keanecjy maybe you could parse the skill title to [str1, str2, str3] then check if any of the sub matches your requirements? something to that effect

Reply:
Good idea. Will update.